docs/concepts/dataclasses.md

```diff
@@ -100,52 +100,45 @@ print(TypeAdapter(User).json_schema())
 """
 ```
 
-`pydantic.dataclasses.dataclass`'s arguments are the same as the standard decorator, except one extra
-keyword argument `config` which has the same meaning as [model_config][pydantic.config.ConfigDict].
-
-!!! warning
-    After v1.2, [The Mypy plugin](../integrations/mypy.md) must be installed to type check _pydantic_ dataclasses.
-
-For more information about combining validators with dataclasses, see
-[dataclass validators](validators.md#dataclass-validators).
+The Pydantic `@dataclass` decorator accepts the same arguments as the standard decorator, with the addition
+of a `config` parameter.
 
 ## Dataclass config
 
-If you want to modify the `config` like you would with a `BaseModel`, you have two options:
+If you want to modify the configuration like you would with a [`BaseModel`][pydantic.BaseModel], you have two options:
 
-* Apply config to the dataclass decorator as a dict
-* Use `ConfigDict` as the config
+* Use the `config` argument of the decorator.
+* Define the configuration with the `__pydantic_config__` attribute.
 
 ```py
 from pydantic import ConfigDict
 from pydantic.dataclasses import dataclass
 
 
-# Option 1 - use directly a dict
-# Note: `mypy` will still raise typo error
-@dataclass(config=dict(validate_assignment=True))  # (1)!
+# Option 1 -- using the decorator argument:
+@dataclass(config=ConfigDict(validate_assignment=True))  # (1)!
 class MyDataclass1:
     a: int
 
 
-# Option 2 - use `ConfigDict`
-# (same as before at runtime since it's a `TypedDict` but with intellisense)
-@dataclass(config=ConfigDict(validate_assignment=True))
+# Option 2 -- using an attribute:
+@dataclass
 class MyDataclass2:
     a: int
+
+    __pydantic_config__ = ConfigDict(validate_assignment=True)
 ```
 
-1. You can read more about `validate_assignment` in [API reference][pydantic.config.ConfigDict.validate_assignment].
+1. You can read more about `validate_assignment` in the [API reference][pydantic.config.ConfigDict.validate_assignment].
 
 !!! note
-    Pydantic dataclasses support [`extra`][pydantic.config.ConfigDict.extra] configuration to `ignore`, `forbid`, or
-    `allow` extra fields passed to the initializer. However, some default behavior of stdlib dataclasses may prevail.
-    For example, any extra fields present on a Pydantic dataclass using `extra='allow'` are omitted when the dataclass
-    is `print`ed.
+    While Pydantic dataclasses support the [`extra`][pydantic.config.ConfigDict.extra] configuration value, some default
+    behavior of stdlib dataclasses may prevail. For example, any extra fields present on a Pydantic dataclass with
+    [`extra`][pydantic.config.ConfigDict.extra] set to `'allow'` are omitted in the dataclass' string representation.
 
 ## Nested dataclasses
```

Comment:
Can we simplify this section and the following one with a single _Similarly to Pydantic models, nested dataclasses and generics are supported_ (and we refer to the relevant model documentation)? Imo the added value is quite low here, we would expect the examples to work anyway

Reply:
I'm fine with this as long as we link to model docs. Maybe we could add a few other notes as well like json schema, etc.